subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022 René Beckmann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt_sn_transport_udp.c
+ *
+ * @brief MQTT-SN UDP transport.
+ */
+
+#include <zephyr/net/mqtt_sn.h>
+
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <logging/log.h>
+LOG_MODULE_DECLARE(net_mqtt_sn, CONFIG_MQTT_SN_LOG_LEVEL);
+
+static int tp_udp_init(struct mqtt_sn_transport *transport)
+{
+	struct mqtt_sn_transport_udp *udp = UDP_CONTAINER(transport);
+	int err;
+
+	udp->sock = zsock_socket(PF_INET, SOCK_DGRAM, 0);
+	if(udp->sock < 0) {
+		return errno;
+	}
+
+	err = zsock_connect(udp->sock, &udp->gwaddr, udp->gwaddrlen);
+	if (err < 0) {
+		return errno;
+	}
+
+	return 0;
+}
+
+static int tp_udp_msg_send(struct mqtt_sn_client *client, struct mqtt_sn_msg *msg,
+	     const void *recipient)
```

Comment:
The `recepient` field does not seem to be used, what's its purpose?

Reply:
removed